qa/workunits/cephtool/test.sh

```diff
@@ -1466,6 +1466,50 @@ function test_mon_osd()
 	expect_false ceph osd set $f
 	expect_false ceph osd unset $f
   done
+
```

Comment:
This test is a good start, but not sufficient. Before you do this positive test, first establish a negative test. That is, in this case, make sure that the flags aren't already set before you presume they are set. The idea is that, in formal logic, the last test implies the first has passed, but none of them with pass without all of them passing.

Reply:
I can't tell whether you're talking about the excerpt above or the one below, but I have the feeling it's about the one above. If so, those tests are from a previous patch that removed support for all those flags -- in which case both setting and unsetting them should be impossible.

If on the test below, at this point of the test we know those flags are not set (at least, not by this test) given above we're actually going through all the flags and both setting and unsetting them (therefore, at least noin, noout, and nodown should be unset). If they are indeed set, then something in between ought to have set them, at which point this whole test is moot anyway given we are not able to control external parties from messing with the cluster being tested.

Reply:
I'm talking about the test code below my comment. Right, we know they're not set by test. I'm just pointing out that this will not catch a regression caused by, for instance, changing a default value. If that's ok, +1.